internal/anti_cheat_quantifiers.go

```diff
@@ -0,0 +1,22 @@
+package internal
+
+import (
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testQuantifiersAsAntiCheat(stageHarness *test_case_harness.TestCaseHarness) error {
+	testCases := []AntiCheatTestCase{
+		{
+			Pattern:          "fo{1,2}bar",
+			Input:            "foobar",
+			ExpectedExitCode: 0,
+		},
+		{
+			Pattern:          "cat",
```

Comment:
@andy1li just reviving my previous comment here: 

> @andy1li I worry that we'll get false positives here. What if a program is implemented poorly such that it returns 0 on unknown inputs?

Even with this new test, wouldn't we still see problems? It's expected that almost all user implementations will handle a simple pattern correctly?

Reply:
To recap the requirement here, we need a set of tests that: 

- (a) are unlikely to ALL pass against a regular user implementation
- (b) are very likely to ALL pass against an off-the-shelf regex implementation

Reply:
Got somewhat carried with all the code refactoring, but forgot the most essential test cases. 😭

Let me ponder this a bit more.